apps/studio/components/interfaces/Settings/Vault/Secrets/EditSecretModal.tsx

```diff
@@ -96,25 +112,22 @@ const EditSecretModal = ({ selectedSecret, onClose }: EditSecretModalProps) => {
         onSubmit={onUpdateSecret}
       >
         {({ isSubmitting, resetForm }: any) => {
-          // [Alaister] although this "technically" is breaking the rules of React hooks
-          // it won't error because the hooks are always rendered in the same order
           // eslint-disable-next-line react-hooks/rules-of-hooks
-          useEffect(() => {
-            if (selectedSecret !== undefined && selectedSecret.decryptedSecret === undefined) {
-              fetchSecretValue()
+          const { isLoading: isLoadingSecretValue } = useVaultSecretDecryptedValueQuery(
+            {
+              projectRef: project?.ref!,
```

Comment:
i'm guessing this needs the connection string as well? the request here throws 400 a few times before rendering properly